File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
from heapq import heappop, heappush, heapify
def solution(s, k):
    heapify(s)
    answer = 0
    while s[0] < k:
        s1 = heappop(s)
        if not s:
            return -1
        s2 = heappop(s)
        heappush(s, s1 + s2 * 2)
        answer += 1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int sc : scoville) {
            pq.add(sc);
        }
        
        int answer = 0;
        
        while (pq.peek() < K) {
            int first = pq.poll();
            if (pq.isEmpty()) {
                return -1;
            }            
            pq.add(first + pq.poll()*2);
            answer++;
        }
        return answer;   
    }
}